extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamInfoItemExtractor.java

```diff
@@ -225,7 +255,26 @@ private boolean isPremium() {
                     return true;
                 }
             }
-        } catch (Exception ignored) {}
+        } catch (Exception ignored) {
+        }
         return false;
     }
+
+    private boolean isPremiere() {
+        return videoInfo.has("upcomingEventData");
```

Comment:
Are you sure this works? I had issues with using `has()` for JsonObjects, so I've been doing `getObject() != null`. Otherwise this PR looks fine to me.

Reply:
What issues did you have with `has(String key)`? I've looked at the source code of that method and it returns `hashmap.contains(key);` So there should not be issues with that.
Btw. we use an outdated version of nanojson. We might want to update it as the new version promises to be faster.

Reply:
It returning `false` when that object exists, but from a quick local test, that doesn't seem to happen, strange…